### PR TITLE
Add support for Navigation.program

### PR DESCRIPTION
**Pull request by avh4**
Lets you create a `TestContext` from parts of a program with navigation.

### CODE REVIEW

**Comment by avh4** [comment]
Review from @brookeangel ?

File: src/Navigation/Extra.elm [hunk 1]
```diff
@@ -0,0 +1,26 @@
+module Navigation.Extra exposing (locationFromString)
+
+{-| TODO: this module should implement the algorithm described at
+<https://www.w3.org/TR/2012/WD-url-20120524/>
+-}
+
+import Navigation
+
+
+{-| Returns `Nothing` if the given string is not a valid absolute URL.
```
Comment (brookeangel) [inline src/Navigation/Extra.elm]: Might be good to add an example of an absolute URL here

File: tests/TestContextTests.elm [hunk 2]
```diff
@@ -56,4 +59,68 @@ all =
...
+                    (Json.Decode.field "x" Json.Decode.string)
+                    .pathname
+                    { init = \flags location -> "<INIT:" ++ location.pathname ++ ":" ++ flags ++ ">"
+                    , update = testUpdate
+                    , view = testView
+                    }
+                    "https://example.com/path"
+                    """{"x": "fromJson"}"""
+                    |> TestContext.expectModel (Expect.equal "<INIT:/path:fromJson>")
+                    |> TestContext.done
```
Comment (brookeangel) [inline tests/TestContextTests.elm]: 🎉 Woohoo, looks good! Do you plan on testing any of the error cases? (I'm okay with no tests, just wondering!)

File: src/Navigation/Extra.elm (hunk 1 above)
Comment (avh4) [reply src/Navigation/Extra.elm]: Fixed in https://github.com/avh4/elm-program-test/pull/1/commits/d429b61182e83a43d114111f0830c311dce6d003

File: tests/TestContextTests.elm (hunk 2 above)
Comment (avh4) [reply tests/TestContextTests.elm]: Maybe eventually when it's more possible to control what the error messages look like.  Basically all the tests failed before they were implemented, so I felt comfortable with that.

**Pull request merged**